Migration-V2.md

```diff
@@ -0,0 +1,67 @@
+# Migrating to version 2.0.0
+
+In order to move from versions `1.x` to version `2.0.0`, there are some breaking changes that need to be accounted for. This guide will address overall changes happening to all service packages, and then call out important changes to individual packages.
+
+## Overall Changes to all Packages
+
+### Breaking `go-sdk-core` version change
+
+For all packages, we now import the `go-sdk-core/v4` module as a dependency. This replaces `go-sdk-core`. Any code that previously imported the Go Core should replace their existing imports with:
+
+```
+"github.com/IBM/go-sdk-core/v4/core"
+```
+
+### Breaking Version Date Type Change
+
+For all packages, the `Version` property on the service struct is now a `*String`. It had previously been a `String` which was provided as a literal. The Go SDK Core has a convenience method to provide this string pointer:
+
+```go
+service, err = assistantv1.NewAssistantV1(&assistantv1.AssistantV1Options{
+	Version:     core.StringPtr("2020-04-01"),
+	ServiceName: "assistant",
+ })
+```
+
+### Breaking Changes to model setter functions
+
+In the `1.x` branch of the SDK, setter methods were provided for almost all model structs for dealing with data from the service. Version `2.0.0` makes significant breaking changesin how setters are handled for these structs.
+
+Specifically, now the only structs that have setters for their properties are structs with the naming convention `{ServiceOperation}Options`. These are called `Options Models` and are typically the top level struct that is passed into a function. These structs have setter methods exposed that allow the individual properties within them to be set.
+
+For lower level models underneath these options models, they had setter methods for each individual property. As this bloated the code surface and provided little value compared to setting the property directly, we have decided in version `2.0.0` to remove these setters.
+
+If your code editor calls out that a setter method you are using has been removed, the appropriate way to update is to set the property directly.
+
+### Providing custom `Context` to functions
+
+Version `2.0.0` of the Go SDK now adds alternate versions to all service functions, named `{function}WithContext`. This function calls the expected service function, but accepts a `Context` as the first parameter. This allows you to provide a context that you would like the function to execute in.
+
+Internally, the Go SDK uses these functions for all operations, but the regular function provides the background context as a default.
+
+### Enabling `gzip` compression
+
+All service structs now expose a function `GetEnableGzipCompression()` and `SetEnableGzipCompression(enableGzip bool)`. These methods allow you to enable `gzip` compression within the service methods.
```

Comment:
I would NOT mention these functions unless you know for a fact that watson services allow the client to send request bodies that are gzip-compressed.   This feature is typically enabled by setting the `x-gzip-compress-requests` extension in the API definition.  If that extension is set, then the service is generated such that gzip compression is enabled by default.  If it's not set, then we still generate the "enable gzip" function so the user could explicitly enable it, but it's not enabled by default.
So my recommendation is to not mention this at all.